lib/src/git.rs

```diff
@@ -1423,32 +1616,50 @@ pub fn rename_remote(
             is_git_remote_ref(&[old_remote_name, new_remote_name]),
         )?;
     }
+
     Ok(())
 }
 
 pub fn set_remote_url(
-    git_repo: &git2::Repository,
+    git_repo: gix::Repository,
     remote_name: &str,
     new_remote_url: &str,
 ) -> Result<(), GitRemoteManagementError> {
     if remote_name == REMOTE_NAME_FOR_LOCAL_GIT_REPO {
         return Err(GitRemoteManagementError::RemoteReservedForLocalGitRepo);
     }
 
-    // Repository::remote_set_url() doesn't ensure the remote exists, it just
-    // creates it if it's missing.
-    // Therefore ensure it exists first
-    git_repo.find_remote(remote_name).map_err(|err| {
-        if is_remote_not_found_err(&err) {
-            GitRemoteManagementError::NoSuchRemote(remote_name.to_owned())
-        } else {
-            GitRemoteManagementError::InternalGitError(err)
-        }
-    })?;
+    let Some(result) = git_repo.try_find_remote(remote_name) else {
+        return Err(GitRemoteManagementError::NoSuchRemote(
+            remote_name.to_owned(),
+        ));
+    };
+    let remote = result.map_err(GitRemoteManagementError::from_git)?;
+
+    let mut new_remote = git_repo
+        .remote_at(new_remote_url)
+        .map_err(GitRemoteManagementError::from_git)?
+        .with_fetch_tags(remote.fetch_tags());
+    for direction in [gix::remote::Direction::Fetch, gix::remote::Direction::Push] {
```

Comment:
Perhaps, the old git2 implementation wouldn't update the "push" URL.

Reply:
This isn’t URLs, it’s refspecs. This is just copying the data from the old remote wholesale (after creating a new one with the right URL), because for some reason the `gix` API doesn’t allow you to set the URL of an existing `Remote`, only create a new one. I assume `libgit2` doesn’t have that limitation and updates the fetch URL in place.

Reply:
Right, I didn't notice that. It might be easier to use the config API, but I'm not sure.

Reply:
I’ve abstracted this out into a `remote_fetch_url` function to shim the missing `gix` API and added an explanatory comment for the data copying. Since @Byron says that this is just a missing API on the `gix` side, I think doing it this way is nicest, as once it grows the missing API the `set_remote_url` code will remain very simple. `gix` implements a fair amount of logic to handle existing config sections, so I think it’s best to use the higher‐level API when possible, although I don’t know if it’s specifically relevant in this place.